### ISSUE TITLE
Full-screen Pharo 9 misbehaving on Mint 20.3 with Cinnamon DT.

### ISSUE THREAD

**Issue by cajavi** [guillep/pharo-core#10965]
Have yesterday installed first time ever PharoLauncher-linux-2.3-x64.zip and used it to download and run Pharo 9.0 64-bit stable.  It seemed to run fine.  I clicked on the black display option and that worked.  I clicked on the full-screen option on the welcome window, and Pharo would no longer work properly.  I made no other changes, did no programming.  The cursor was offset, so I was able to guess the offset and quit the program without saving.  Tried again and went into settings to check fullscreen instead of via the welcome window.  This made things much worse.  Totally misbehaved under full-screen setting (windows off-screen, could see cursor at all, and who knows what else).  Was not able to safely exit, and I shut down my system.
The system I am running is Linux Mint 20.3 with Cinnamon desktop.  It's running on an HP EliteDesk 800 G3 SFF with 32GB memory and an SSD for the OS.  Not sure what else I can tell you that would be relevant.  
The bottom line is that I did very little but download and run, and it misbehaves when set to fullscreen.  If you have further questions about the system environment please ask, thanks.
PS.  I am not familiar with Github... I joined to be able to report this problem.

**Comment by johnhwoods** [comment]
Also problematic on Ubuntu 21 with KDE Plasma.  Suggest urgent workaround is to disable fullscreen option for Linux platform.

**Comment by johnhwoods** [comment]
Also tested on P10, issue persists.

**Comment by cajavi** [comment]
Thanks to inspiration from johnhwoods testing on other os, decided to test P9 on Windows 10 (pro) via new download and run of launcher, same physical machine as I used for the initial problem report.   The full-screen problem did NOT occur on windows 10 (I wonder if that is where most Pharo development and testing takes place?).

**Comment by cajavi** [comment]
Not realizing what is was doing I accidentally closed this thread when I was trying to submit my latest test description instead.  Hope it can be reopened.

**Comment by johnhwoods** [comment]
I think they're all using Macs - look at the window button placement!  I've tried another couple of windows managers on Linux again without success.  Sometimes you do get a transient fullscreen.  Unfortunately this is all stuff at a much lower system level than any of my 'expertise' such as it is.

**Comment by johnhwoods** [comment]
Hi @cajavi , I presume you are getting behaviour similar to this https://youtu.be/AsHMNTuAoa4

<blockquote><div><strong><a href="https://youtu.be/AsHMNTuAoa4">https://www.youtube.com/watch?v=AsHMNTuAoa4&feature=youtu.be</a></strong></div></blockquote>

**Comment by StephanEggermont** [comment]
Also occurs on Ubuntu 20.04 LTS. And no, not all of us use Macs. I'm fully three-platform. Looks like an underlying library problem (SDL). Disabling in the image is only an option if we determine that the underlying problem is unlikely to be fixed soon

**Comment by johnhwoods** [comment]
@cajavi @StephanEggermont - any chance you could join in on the Linux channel on Pharo Discord?  I cannot agree that 'disabling in the image is only an option .." because fullscreen isn't a big deal and the fullscreen failure is about the biggest deal I can imagine.  the existence of a quick workaround now is not going to prevent anyone fixing the underlying problem.

**Comment by cajavi** [comment]
@johnhwoods As to your request to join in on Discord, I'm sorry I've been having much trouble getting into my discord account, partly due to captcha, and maybe because not used in over two years.  Tried the lost password send email option also, but it did not work either, despite having recently received a Discord administrative email to that email account.  So unfortunately, for now, I am replying to you here.
To reply re whether the behaviour is similar to your Ubuntu/KDE video:  it is not.  First there was just a cursor offset problem, then on 2nd launch, no cursor at all, and windows partially off-screen, not capable of doing anything with the system, as if damage had been caused in the first run and affected second run.  In your video, there are flashing windows which appear never to get focus.  Still I suspect the underlying cause is the same.  Probably there is memory overwrite going on somewhere causing random damage.
Since then I've tried to help by running in Win Pro 10 (seemed to be ok for fullscreen) and also in a vm running a live Manjaro Linux distro, but the run crashed entirely and I lost the log of that.  All I remember is that there was a stack overflow at the end (which would cause random damage I suppose), but this may have been and likely was an entirely different issue.  
Then I tried back in my actual Mint OS to trace the bug itself by putting in a breakpoint and tried stepping through the full-screen switch.  I probably did a poor job of this, but my latest try of that had a final effect of total freeze-up of my system, a black screen, and no response at all from keyboard or mouse.  Was forced to cycle the power switch.  I probably just have insufficient skills to be helping out.  I've not yet even identified where the actual crash/freeze occurred (if there is even such a spot as that could be random).  If I do continue testing I think I should I perhaps use a VM because I do not want to keep freezing up (and possibly damaging) my Mint OS, which is my daily driver.  I hope it is not already damaged, though it seems okay after reboot.

As for your asking my opinion on removing the full-screen option, I tend to agree with you that this problem is a major bug, even if just from an optics/PR perspective, since major Linux distros are affected, and potentially affects any usage at all.  Given how I felt, I can imagine how it would totally discourage a new user of Pharo!  So this would hopefully either be fixed quickly, or as you suggest the feature be disabled.  Personally I would prefer to just have a warning added to the setting requiring a ok/orCancel to proceed (warning to use at own risk as may not work on your Linux distro), as this would allow anyone to continue using the fullscreen or even testing it on any distro they wish, for Linux versions only I presume (I cannot test a MAC).
 
As I've mentioned I'm new to Github and to issues contribution, but I think it might be handy if an issue could be evaluated as to its urgency when it is first created (perhaps there is something informal in place for that?).  Perhaps the creator of the issue could initially evaluate its urgency and this could then be overridden by someone with more info/authority from the project.
Did not mean to be chatting here, but as I said, I'm having trouble getting into Discord.  And hopefully some of the above may be of some help (though I doubt it).  
P.S. I will keep trying to join on Discord.
